Crashlytics/Crashlytics/Rollouts/CrashlyticsRemoteConfigManager.swift

```diff
@@ -0,0 +1,61 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import FirebaseRemoteConfigInterop
+import Foundation
+
+protocol CrashlyticsPersistentLog: NSObject {
+  func updateRolloutsStateToPersistence(rolloutAssignments: [RolloutAssignment])
+}
+
+@objc(FIRCLSRemoteConfigManager)
+public class CrashlyticsRemoteConfigManager: NSObject {
+  public static let maxRolloutAssignment = 128
+  public static let maxParameterValueLength = 256
+
+  var remoteConfig: RemoteConfigInterop
+  public private(set) var rolloutAssignment: [RolloutAssignment] = []
+  weak var persistenceDelegate: CrashlyticsPersistentLog?
+
+  @objc public init(remoteConfig: RemoteConfigInterop) {
+    self.remoteConfig = remoteConfig
+  }
+
+  @objc public func updateRolloutsState(rolloutsState: RolloutsState) {
+    rolloutAssignment = validateRolloutAssignment(assignments: Array(rolloutsState.assignments))
+  }
+}
+
+private extension CrashlyticsRemoteConfigManager {
+  func validateRolloutAssignment(assignments: [RolloutAssignment]) -> [RolloutAssignment] {
+    var validatedAssignments = assignments
+    if assignments.count > CrashlyticsRemoteConfigManager.maxRolloutAssignment {
+      validatedAssignments =
+        Array(assignments[..<CrashlyticsRemoteConfigManager.maxRolloutAssignment])
```

Comment:
Maybe log a warning here? Or a debug-level message?

Reply:
We have internal helper for logging provided by Core. But since the swift target does not have any dependencies, I will use `debugLog()` from Swift library. This probably will apply to interop as well